src/psyclone/tests/psyir/tools/dependency_tools_test.py

```diff
@@ -172,25 +175,115 @@ def test_arrays_parallelise(parser):
            "and jj and can therefore not be parallelised" \
            in dep_tools.get_all_messages()[0]
 
 
+# -----------------------------------------------------------------------------
+def test_array_access_consistent(parser):
+    '''Tests the array checks of can_loop_be_parallelised.
+    '''
+    # pylint: disable=too-many-locals
+    reader = FortranStringReader('''program test
+                                 integer ji, jj, jk
+                                 integer, parameter :: jpi=5, jpj=10
+                                 real, dimension(jpi,jpi) :: a, b, c
+                                 do jj = 1, jpj   ! loop 0
+                                    do ji = 1, jpi
+                                       a(ji, jj) = -1.0d0
+                                     end do
+                                 end do
+                                 do jj = 1, jpj   ! loop 1
+                                    do ji = 1, jpi
+                                       a(ji, jj) = b(ji, jj)
+                                       c(ji, jj) = a(ji, jj)
+                                     end do
+                                 end do
+                                 do jj = 1, jpj   ! loop 2
+                                    do ji = 1, jpi
+                                       a(jj, ji) = b(ji, jj)
+                                     end do
+                                 end do
+                                 end program test''')
+    prog = parser(reader)
+    psy = PSyFactory("nemo", distributed_memory=False).create(prog)
+    loops = psy.invokes.get("test").schedule
+    dep_tools = DependencyTools(["levels", "lat"])
+
+    var_info0 = VariablesAccessInfo(loops[0])
+    var_info1 = VariablesAccessInfo(loops[1])
+    jj_symbol = loops.scope.symbol_table.lookup("jj")
+
+    # Test 1: provide access pattern of different variables:
+    sig_a = Signature("a")
+    sig_b = Signature("b")
+    sig_c = Signature("c")
+    with pytest.raises(InternalError) as err:
+        _ = dep_tools.array_accesses_consistent(jj_symbol,
+                                                [var_info0[sig_a],
+                                                 var_info1[sig_b],
+                                                 var_info1[sig_c]])
+    assert "Inconsistent signature provided in 'array_accesses_consistent'. " \
+           "Expected all access to be for 'a', but also got 'b,c'."\
+           in str(err.value)
+
+    # Test 2: provide a consistent list of accesses.
+    # Check number of messages and returned access array for correctness.
+    a_access_1st_loop = var_info0[sig_a]
+    a_access_2nd_loop = var_info1[sig_a]
+    all_ind = []
+    consistent = dep_tools.array_accesses_consistent(jj_symbol,
+                                                     [a_access_1st_loop,
+                                                      a_access_2nd_loop],
+                                                     all_ind)
+    assert consistent
```

Comment:
Please add `is True` just to make sure that the return value is what we expect.

Reply:
OK, that's new to me. I've updated a lot of tests to use `is` now.

Reply:
Thanks Joerg - it's almost paranoia but it doesn't hurt to be careful :-)